types/fault_set.go

```diff
@@ -0,0 +1,37 @@
+package types
+
+import (
+	"fmt"
+
+	cbor "github.com/ipfs/go-ipld-cbor"
+)
+
+func init() {
+	cbor.RegisterCborType(FaultSet{})
```

Comment:
not 100% but I think this is unnecessary because you're not defining a custom atlas for this type

Reply:
Yeah, apparently you do, you get an error, unless we enable auto-atlasing.
```
panic: missing an atlas entry describing how to marshal type types.FaultSet (and auto-atlasing for structs is not enabled)
```
Looking at the source it is [unclear](https://github.com/polydawn/refmt/blob/10eebc3c830751b23b9236fc4df06a2d95e0f894/obj/marshalSlab.go#L146) whether this would even work..